src/main/java/resources/NasaResource.java

```diff
@@ -0,0 +1,55 @@
+package resources;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.Response.Status;
+import millom.sandbox.CustomException.InvalidWeatherException;
+import millom.sandbox.service.WeatherService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/nasaapi")
+public class NasaResource {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(NasaResource.class);
+
+  private final WeatherService weatherService;
+
+  public NasaResource(WeatherService weatherService) {
+    this.weatherService = weatherService;
+  }
+
+  public NasaResource() {
+    this(new WeatherService());
+  }
+
+  @Path("/mars/weather/")
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getMarsWeather(@QueryParam("feed") String feed,
+      @QueryParam("feedtype") String feedType,
+      @QueryParam("version") float version, @QueryParam("category") String category) {
+    try {
+      LOGGER.info(
+          String.format("Task-3: Success message from Nasa API -- %s class and  get method: %s()",
+              Thread.currentThread().getStackTrace()[1].getClassName(),
+              Thread.currentThread().getStackTrace()[1].getMethodName()));
+      return Response.
+          status(Response.Status.OK)
+          .entity(weatherService.getWeatherMapping(feed, feedType, version, category))
```

Comment:
Not necessary to do anything about it here, but in general it is always a good idea to sanitize input variables
before they are used. Especially if they are forwarded upstream (to the Nasa API).

But in this case I think it is OK to leave things as they are.

Reply:
Tried to sanitise the input variables by validating input variables with some assumptions.

```
 if (feed == null || feed.length() == 0 || !feed.equals("weather")) {
      return getErrorResponse("feed");
    }
    if (feedType == null || feedType.length() == 0 || !feedType.matches("json|xml|text")) {
      return getErrorResponse("feedType");
    }
    String versionToString = String.valueOf(version);

    if (versionToString == null || versionToString.length() < 2 || !versionToString.matches(
        "^(\\d)*(?:\\.\\d)?$")) {
      return getErrorResponse("version");
    }
    if (category == null || category.length() < 2 || category.length() > 5) {
      return getErrorResponse("category");
    }
```

```
private Response getErrorResponse(String field) {
    return Response.
        status(Status.INTERNAL_SERVER_ERROR)
        .entity(String.format("The input values are not valid for %s.", field))
        .build();
  }
```